modules/ui/map_in_map.js

```diff
@@ -1,327 +1,297 @@
-//            .merge(wrap);
-//
-//        // reflow warning: Hardcode dimensions - currently can't resize it anyway..
-//        _dMini = [200,150]; //utilGetDimensions(wrap);
-//        _cMini = vecScale(_dMini, 0.5);
-//
-//        context.map()
-//            .on('drawn.map-in-map', function(drawn) {
-//                if (drawn.full === true) {
-//                    redraw();
-//                }
-//            });
-//
-//        redraw();
-//
-//        context.keybinding()
-//            .on(t('background.minimap.key'), toggle);
-//    }
-//
-//    return mapInMap;
-//}
+import * as PIXI from 'pixi.js';
+import { select as d3_select } from 'd3-selection';
+import { zoom as d3_zoom, zoomIdentity as d3_zoomIdentity } from 'd3-zoom';
+import {
+  Projection,
+  geoScaleToZoom,
+  geoZoomToScale,
+  vecScale,
+  vecSubtract,
+} from '@id-sdk/math';
+import { t } from '../core/localizer';
+import { utilSetTransform } from '../util';
+import { PixiScene } from '../pixi/PixiScene';
+import { PixiLayerBackgroundTiles } from '../pixi/PixiLayerBackgroundTiles';
+
+
+export function uiMapInMap(context) {
+  function mapInMap(selection) {
+    var projection = new Projection();
+    var zoom = d3_zoom()
+      .scaleExtent([geoZoomToScale(0.5), geoZoomToScale(24)])
+      .on('start', zoomStarted)
+      .on('zoom', zoomed)
+      .on('end', zoomEnded);
+
+    var wrap = d3_select(null);
+    var tiles = d3_select(null);
+    var viewport = d3_select(null);
+
+    var _isTransformed = false;
+    var _isHidden = true;
+    var _skipEvents = false;
+    var _gesture = null;
+    var _zDiff = 6; // by default, minimap renders at (main zoom - 6)
+    var _dMini; // dimensions of minimap
+    var _cMini; // center pixel of minimap
+    var _tStart; // transform at start of gesture
+    var _tCurr; // transform at most recent event
+    var _timeoutID;
+
+
+    function zoomStarted() {
+      if (_skipEvents) return;
+      _tStart = _tCurr = projection.transform();
+      _gesture = null;
+    }
+
+
+    function zoomed(d3_event) {
+      if (_skipEvents) return;
+
+      var x = d3_event.transform.x;
+      var y = d3_event.transform.y;
+      var k = d3_event.transform.k;
+      var isZooming = k !== _tStart.k;
+      var isPanning = x !== _tStart.x || y !== _tStart.y;
+
+      if (!isZooming && !isPanning) {
+        return; // no change
+      }
+
+      // lock in either zooming or panning, don't allow both in minimap.
+      if (!_gesture) {
+        _gesture = isZooming ? 'zoom' : 'pan';
+      }
+
+      var tMini = projection.transform();
+      var tX, tY, scale;
+
+      if (_gesture === 'zoom') {
+        scale = k / tMini.k;
+        tX = (_cMini[0] / scale - _cMini[0]) * scale;
+        tY = (_cMini[1] / scale - _cMini[1]) * scale;
+      } else {
+        k = tMini.k;
+        scale = 1;
+        tX = x - tMini.x;
+        tY = y - tMini.y;
+      }
+
+      if (_gesture === 'pan') {
+        utilSetTransform(tiles, tX, tY, scale);
```

Comment:
So it turns out that setting the transform of the entire canvas isn't great, as this also skootches the bounding box around with it. I will rewire this to actually modify the pixi container offsets for just the imagery tile layer.

Reply:
See my later commits, where I just reposition the container!